### ISSUE TITLE
Running multiple parameter settings in parallel for omicsintegrator1 creates race conditions

### ISSUE THREAD

**Issue by sverchkov** [Reed-CompBio/spras#59]
Observed using omicsintegrator1 with singularity, might be an issue in other settings too.

Giving snakemake multiple cores (`--cores` parameter) and running multiple parameter settings creates a situation where multiple `reconstruct` rules are run concurrently. Because omics integrator 1 temporarily uses some fixed file names in the output directory, this creates a race condition where the different runs all try to write/move the same files.

The particular error I got consistently is that the `oi1_optimalForest.sif` file is not found when trying to rename it. Other files might also be affected.

The workaround is to always call snakemake with `--cores 1`.

**Comment by agitter** [comment]
Thanks for the report @sverchkov.  Which commit of SPRAS are you running?  This issue sounds similar to #31 and was hopefully addressed by #49.

**Comment by sverchkov** [comment]
Ah, you're right, this was with an older version. All good on the latest main branch.

**Comment by sverchkov** [comment]
Sorry, I spoke too soon. Now getting that error with 1 core too. I'll make another issue.
